kale/workflow_objects.py

```diff
@@ -366,10 +396,36 @@ def draw_dag(self, layout=None):
         toolbar = bq.Toolbar(figure=fig)
 
         return ipw.VBox([fig, toolbar])
+    
+    def check_selection_for_jumps(self):
+        """
+        It's okay for a selcted task to have a parent or child
+        which is not selected. But it's not okay to have
+        a generational gap in selection.
+        """
+        pass
 
     def gen_subdag(self):
         """Return DAG containing only steps which are to be run. (Not yet implemented.)"""
-        return self.dag
+        subdag = networkx.DiGraph()
+
+        if self._bqgraph.selected is None or len(self._bqgraph.selected) == 0:
+            return self.dag
+        else:
+            self.check_selection_for_jumps()
+
+            for node in self.dag.nodes():
```

Comment:
this section in gen_subdag should have comments describing the intent of the code for future reference

Reply:
Wondering if you can iterate over self._bqgraph.selected instead, then check for ancestors and add edges?

Reply:
Good idea - I changed this